src/test/ui/nll/dont-print-desugared.rs

```diff
@@ -0,0 +1,29 @@
+// Test that we don't show variables with from internal desugarings
+
+// edition:2018
```

Comment:
this is only testing behavior for 2018 edition.

Can you add a similar test that checks the 2015 edition as well? (Or remove the "edition" mark from this file...?) Or is there some reason it makes sense to only test this against the 2018 edition? (Maybe I missed some new policy regarding testing in terms of whether to encode an edition or not.)

Reply:
`async fn` is >=2018 edition only

Reply:
I've split the `async fn` case into its own test to maybe make this more obvious.